### PR TITLE
feat: allow bi-directional data transfers

### PR DESCRIPTION
**Pull request by paullatzelsperger**
## What this PR changes/adds

this PR introduces the capability on provider dataplanes to create "response channel EDRs": data planes that support this can expose a "response channel" through which consumers can tunnel back control information

## Why it does that

_Briefly state why the change was necessary._

## Further notes

_List other areas of code that have changed but are not necessarily linked to the main feature. This could be method
signature changes, package declarations, bugs that were encountered and were fixed inline, etc._

## Linked Issue(s)

Closes # <-- _insert Issue number if one exists_

_Please be sure to take a look at the [contributing guidelines](https://github.com/eclipse-edc/.github/blob/main/CONTRIBUTING.md#submit-a-pull-request) and our [etiquette for pull requests](https://github.com/eclipse-edc/.github/blob/main/contributing/pr_etiquette.md)._

### CODE REVIEW

File: core/control-plane/control-plane-transfer/src/main/java/org/eclipse/edc/connector/controlplane/transfer/flow/TransferTypeParserImpl.java [hunk 1]
```diff
@@ -23,12 +23,19 @@
...
+     * Parses a compose transfer type string into a {@link TransferType}:
+     * {@code DESTTYPE-{PUSH|PULL}(-RESPONSETYPE)}, for example {@code HttpData-PULL/Websocket}
+     *
+     * @param transferType the transfer type string representation.
+     * @return a {@link TransferType}
+     */
     @Override
     public Result<TransferType> parse(String transferType) {
         Optional<Result<TransferType>> parsed = Optional.ofNullable(transferType)
                 .map(type -> type.split("-"))
```
Comment (ndr-brt) [inline core/control-plane/control-plane-transfer/src/main/java/org/eclipse/edc/connector/controlplane/transfer/flow/TransferTypeParserImpl.java:36]: shouldn't it be parsed by `/` and then by `-`?

File: core/control-plane/control-plane-transfer/src/test/java/org/eclipse/edc/connector/controlplane/transfer/flow/TransferTypeParserImplTest.java [hunk 2]
```diff
@@ -60,4 +60,10 @@ void shouldReturnFatalError_whenFormatIsNotCorrect() {
         assertThat(result).isFailed();
     }
 
+    @Test
+    void shouldParseReturnChannel() {
+        var result = parser.parse("DestinationType-PUSH-BackChannelType");
```
Comment (ndr-brt) [inline core/control-plane/control-plane-transfer/src/test/java/org/eclipse/edc/connector/controlplane/transfer/flow/TransferTypeParserImplTest.java:65]: indeed
```suggestion
        var result = parser.parse("DestinationType-PUSH/BackChannelType");
```

File: core/control-plane/control-plane-transfer/src/main/java/org/eclipse/edc/connector/controlplane/transfer/flow/TransferTypeParserImpl.java (hunk 1 above)
Comment (paullatzelsperger) [reply core/control-plane/control-plane-transfer/src/main/java/org/eclipse/edc/connector/controlplane/transfer/flow/TransferTypeParserImpl.java:36]: @wolf4ood was having doubts about the "/" causing troubles with json-ld

File: docs/developer/decision-records/2024-10-24-bidirectional-data-transfers/README.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+The `TransferType` object will be extended to contain another optional lproperty `responseChannelType`, which is encoded
+as follows:
+
+```
+DESTTYPE-FLOW/RESPONSETYPE, e.g. HttpData-PULL/gRpc
+```
+
+In the catalog, every `DataSet` will be represented by `Distributions` both with and without the `responseChannelType`,
+e.g. `HttpData-PULL/gRpc` and `HttpData-PULL`. This keeps consistency with the current data plane selection mechanism.
```
Comment (ndr-brt) [inline docs/developer/decision-records/2024-10-24-bidirectional-data-transfers/README.md:97]: the DR needs to be adapted then

**Review by ndr-brt — APPROVED** [review]

**Pull request merged**